[Marcus Low] iP

marcuslowhuiyu opened the pull request.

Commit:
Add Gradle support

marcuslowhuiyu committed:
Level-1: Greet, Echo, Exit

marcuslowhuiyu committed:
Level-1: Greet, Echo, Exit

marcuslowhuiyu committed:
Level 2. Add, List

marcuslowhuiyu committed:
Level 3. Mark as Done

marcuslowhuiyu committed:
Rename events to tasks

marcuslowhuiyu committed:
Level 4. ToDos, Events, Deadlines

marcuslowhuiyu committed:
A-TextUiTesting

marcuslowhuiyu committed:
Level 5. Handle Errors

marcuslowhuiyu committed:
Level 6. Delete

marcuslowhuiyu committed:
Level-7: Save
Enabled writing to file after each addition of a task
Modified the various tasks to write to the file in proper format
Fixed errors with deadline and event

marcuslowhuiyu committed:
Revert "Level-7: Save"

This reverts commit d7f0622ab154e64000d4c72246c9dfa55dac52fd.

marcuslowhuiyu committed:
Change for loops to enhanced for loops

marcuslowhuiyu committed:
Merge branch 'branch-Level-7' of https://github.com/marcuslowhuiyu/ip into branch-Level-7

# Conflicts:
#	src/main/java/Doris.java

marcuslowhuiyu committed:
Merge branch 'branch-Level-7' of https://github.com/marcuslowhuiyu/ip into branch-Level-7

# Conflicts:
#	src/main/java/Doris.java

marcuslowhuiyu committed:
Level-8: Dates and Times
Add support for localdatetime

marcuslowhuiyu committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Doris.java

seetohjinwei commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,27 @@
+public abstract class Task {
+    protected String description;
+    protected boolean done = false;
```
Perhaps use a variable name that sounds more like a boolean?

seetohjinwei commented on src/main/java/Doris.java:
```diff
@@ -0,0 +1,134 @@
+public class Doris {
+    ArrayList<Task> list = new ArrayList<Task>();
+
+    DateTimeFormatter df = DateTimeFormatter.ofPattern("yyyy-MM-dd hh:mm a");
```
Perhaps use a more descriptive name for this? Its scope seems to be pretty large.

seetohjinwei commented on src/main/java/Doris.java:
```diff
@@ -0,0 +1,134 @@
+import java.io.*;
```
Perhaps list the imported classes explicitly instead?

seetohjinwei commented on src/main/java/Doris.java:
```diff
@@ -0,0 +1,134 @@
+
+    DateTimeFormatter df = DateTimeFormatter.ofPattern("yyyy-MM-dd hh:mm a");
+
+    public void start (){
```
I think you missed out a space here!

seetohjinwei reviewed (commented):
Hi! For coding standard related issues, I only found a few minor issues. Overall, it's pretty good!

marcuslowhuiyu committed:
Refactor code to follow OOP principles

TanPingZhi commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,28 @@
+        } else {
+            return "[E][ ] " + this.description + " (by: " +
+                    this.time.format(DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm a")) + ")";
+        }
```
You can use a ternary operator `done ? "X" : " " `here

TanPingZhi commented on doris/doris.txt:
```diff
@@ -0,0 +1 @@
+D | false | this | 2 Feb 2000
```
you may want to use gitignore for this file

TanPingZhi reviewed (commented):
No problem with naming or layouts
Just small improvements to make

marcuslowhuiyu committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

wxy1203 commented on src/main/java/Task.java:
```diff
@@ -1,12 +1,13 @@
-    private String description;
-    private boolean done = false;
+    protected String description;
+    protected boolean done = false;
```
Nice that you change the fields to be protected :+1:

wxy1203 commented on src/main/java/Doris.java:
```diff
@@ -27,50 +27,66 @@ public void start (){
+                    for (int i = 0; i < list.size(); i++) {
+                        System.out.println(i + 1 + ". " + list.get(i));
+                    }
+                } else if (command.startsWith("mark")) {
```
Maybe too much if else here, my suggestion is to use switch :smiley:

wxy1203 commented on src/main/java/Doris.java:
```diff
@@ -27,50 +27,66 @@ public void start (){
+                    }
+                    String[] commands = command.split(" /at ");
+                    if (commands.length > 2) {
+                        throw new DorisException("Oi can don't talk too much");
```
Nice exception throw :+1:

wxy1203 commented on src/main/java/Doris.java:
```diff
@@ -87,19 +81,17 @@ public void start (){
-            } catch (IOException e) {
-                e.printStackTrace();
             }
         }
```
Really like the printStackTrace for the exception, and this is LGTM :+1:

wxy1203 commented on src/main/java/Event.java:
```diff
@@ -1,18 +1,23 @@
 
-    public Event(String command, String time) {
+    public Event(String command, LocalDateTime time) {
         super(command);
```
There is no documentation in your code, maybe try to take a look at the java documentation to give others a clear idea on the functionality of the classes. 👊

wxy1203 commented on src/main/java/Doris.java:
```diff
@@ -1,10 +1,15 @@
 import java.io.*;
```
This needs to be specific according to the java coding style. :cry:

wxy1203 commented on src/main/java/Doris.java:
```diff
@@ -75,8 +80,8 @@ public void start (){
-                    String[] commands = command.substring(6).split(" /at ");
-                    Event event = new Event(commands[0], commands[1]);
+                    String[] commands = command.split(" /at ");
+                    Event event = new Event(commands[0], LocalDateTime.parse(commands[1], df));
```
Nice that you change all your ways of find the command line. :+1:

wxy1203 reviewed (commented):
Overall, the written code is understandable, but need to pay more attention on the java doc and coding style. 👊

marcuslowhuiyu committed:
Add gradle support

marcuslowhuiyu committed:
Add JUnit Tests

marcuslowhuiyu committed:
Build Jar files

marcuslowhuiyu committed:
Add JavaDocs

marcuslowhuiyu committed:
Follow coding standard

marcuslowhuiyu committed:
Add Find Command

marcuslowhuiyu committed:
Merge branch 'branch-A-CodingStandard'

marcuslowhuiyu committed:
Resolv merge issues

marcuslowhuiyu committed:
Change README.md

marcuslowhuiyu committed:
Correct checkstyle violations

marcuslowhuiyu committed:
Add GUI

marcuslowhuiyu committed:
Add assertion to AddCommand

This is to ensure that AddCommand always correctly adds either a deadline or an event task

Might want to look into how to do this check in a better way in the future

marcuslowhuiyu committed:
Improve code quality

marcuslowhuiyu committed:
no message

marcuslowhuiyu committed:
Add user guide

marcuslowhuiyu committed:
Remove old README

marcuslowhuiyu committed:
no message

marcuslowhuiyu committed:
Update README

Edfernape commented on src/main/java/exception/DorisException.java:
```diff
@@ -0,0 +1,17 @@
+package exception;
```
very neat organization of files!

Edfernape reviewed (commented):
great folder organization

marcuslowhuiyu committed:
Rename OldREADME.md to README.md

marcuslowhuiyu committed:
Rename README.md to OLDREADME.md
